llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -961,23 +961,27 @@ getSizeFromDwarfHeaderAndSkip(BinaryStreamReader &Reader) {
 
 /// Given a list of MCFragments, return a vector with the concatenation of their
 /// data contents.
-/// If any fragment is not an MCDataFragment, an error is returned.
+/// If any fragment is not an MCDataFragment, or the fragment is an
+/// MCDwarfLineAddrFragment and the section containing that fragment is not a
+/// debug_line section, an error is returned.
 static Expected<SmallVector<char, 0>>
-mergeDataFragmentContents(const MCSection::FragmentListType &FragmentList) {
+mergeMCFragmentContents(const MCSection::FragmentListType &FragmentList,
+                        bool IsDebugLineSection = false) {
   SmallVector<char, 0> mergedData;
-  for (const MCFragment &Fragment : FragmentList) {
-    if (const auto *DataFragment = dyn_cast<MCDataFragment>(&Fragment))
-      llvm::append_range(mergedData, DataFragment->getContents());
+  for (const MCFragment &F : FragmentList) {
+    if (F.getKind() == llvm::MCFragment::FT_Data)
+      llvm::append_range(mergedData, cast<MCDataFragment>(F).getContents());
+    else if (F.getKind() == llvm::MCFragment::FT_Dwarf && IsDebugLineSection)
+      llvm::append_range(mergedData,
+                         cast<MCDwarfLineAddrFragment>(F).getContents());
     else
       return createStringError(inconvertibleErrorCode(),
                                "Invalid fragment type");
   }
   return mergedData;
 }
 
-Error MCCASBuilder::splitDebugInfoCUs() {
-  auto *DebugInfoSection =
-      Asm.getContext().getObjectFileInfo()->getDwarfInfoSection();
+Error MCCASBuilder::splitDebugInfoCUs(const MCSection *DebugInfoSection) {
```

Comment:
This change also seems unrelated to the PR?

Reply:
When I rebased, I wanted to make things consistent, so I just added the MCSection as a parameter. If you think we should have a separate PR just to make things consistent, that is fine too, but I didn't see it that way

Reply:
In general I'd just favour minimizing the diff (makes it easier for reviwers) and keeping NFC changes separate